py/mpconfig.h

```diff
@@ -1992,15 +2015,17 @@ typedef double mp_float_t;
 #define MICROPY_MAKE_POINTER_CALLABLE(p) (p)
 #endif
 
-// If these MP_PLAT_*_EXEC macros are overridden then the memory allocated by them
-// must be somehow reachable for marking by the GC, since the native code
-// generators store pointers to GC managed memory in the code.
+// If these macros are defined then the memory allocated by them does not need to be
+// traced by the GC.  But if they are left undefined then the GC heap will be used as
+// the allocator and the memory must be traced by the GC.
 #ifndef MP_PLAT_ALLOC_EXEC
-#define MP_PLAT_ALLOC_EXEC(min_size, ptr, size) do { *ptr = m_new(byte, min_size); *size = min_size; } while (0)
+// Leave as undefined.
+// #define MP_PLAT_ALLOC_EXEC(min_size, ptr, size) do { *ptr = malloc(min_size); *size = min_size; } while (0)
```

Comment:
Not so happy how this bit turned out, needing to leave these undefined to get the correct behaviour.

The point is, if you define this macro then there is different tracking requirements.

Maybe a better way would be something like:
```c
#ifndef MP_PLAT_ALLOC_EXE
// Text tracked by the GC.
#define MP_PLAT_ALLOC_EXEC(min_size, ptr, size) do { *ptr = m_new(byte, min_size); *size = min_size; } while (0)
#define MICROPY_PERSISTENT_CODE_TRACK_FUN_DATA (1)
#define MICROPY_PERSISTENT_CODE_TRACK_BSS_RODATA (0)
#else
// Text not tracked by the GC, explicitly track BSS/rodata.
#define MICROPY_PERSISTENT_CODE_TRACK_FUN_DATA (0)
#define MICROPY_PERSISTENT_CODE_TRACK_BSS_RODATA (1)
#endif
```

Reply:
Yeah, I like this suggestion for how to structure it. A bunch of other #if/#else should simplify out as well.

Reply:
Actually, it wasn't that simple that I could apply my above idea.  But I've adjusted things a bit to keep the existing default definition of `MP_PLAT_ALLOC_EXEC`.

Reply:
New update still a good simplification, I think!